dockerfiles/vault-dcos/Readme.md

```diff
@@ -0,0 +1,114 @@
+# Vault on DC/OS
+
+This is an example of how to run [HashiCorp's Vault](https://github.com/hashicorp/vault) on DC/OS. By default, it's configured in HA mode with ZooKeeper as the backend.
+
+## Running on DC/OS
+
+### Step 1: Launch Vault server
+Save the following JSON as `vault.json`:
+```json
+{
+  "id": "vault",
+  "cpus": 1,
+  "mem": 1000,
+  "ports":[8200],
+  "requirePorts":true,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "brndnmtthws/vault-dcos",
+      "network": "HOST",
+      "forcePullImage": true
+    }
+  },
+  "healthChecks": [{
+      "protocol": "TCP",
+      "portIndex": 0,
+      "timeoutSeconds": 10,
+      "gracePeriodSeconds": 10,
+      "intervalSeconds": 2,
+      "maxConsecutiveFailures": 10
+  }]
+}
+```
+Now launch on Marathon:
+```
+$ dcos marathon app add vault.json
+```
+
+**Note**: you may configure some parameters via environment variables, from the wrapper script [`run-vault`](run-vault). You may set the following environment variables:
+ * VAULT_TLS_KEY: TLS key file contents
+ * VAULT_TLS_CERT: TLS cert file contents
+ * VAULT_CONFIG_HCL: HCL config file contents
+ * VAULT_CONFIG_JSON: JSON config file contents
+
+### Step 2: Initialize the vault
+SSH into one of the DC/OS cluster nodes, and initialize the vault with the following:
+```
+$ docker run -e "VAULT_SKIP_VERIFY=true" -e "VAULT_ADDR=https://vault.marathon.mesos:8200" --entrypoint=vault -t brndnmtthws/vault-dcos init
```

Comment:
Need to reference via virtual IP in VAULT_ADDR: <service-name>.marathon.l4lb.thisdcos.directory:<port>

Or in our case <framework-name>-vault..marathon.l4lb.thisdcos.directory:8200